pkg/api/api.go

```diff
@@ -124,18 +124,18 @@ func (hs *HTTPServer) registerRoutes() {
 	r.Get("/live/pipeline", reqGrafanaAdmin, hs.Index)
 	r.Get("/live/cloud", reqGrafanaAdmin, hs.Index)
 
-	r.Get("/plugins", middleware.CanAdminPlugins(hs.Cfg), hs.Index)
-	r.Get("/plugins/:id/", middleware.CanAdminPlugins(hs.Cfg), hs.Index)
-	r.Get("/plugins/:id/edit", middleware.CanAdminPlugins(hs.Cfg), hs.Index) // deprecated
-	r.Get("/plugins/:id/page/:page", middleware.CanAdminPlugins(hs.Cfg), hs.Index)
+	r.Get("/plugins", middleware.CanAdminPlugins(hs.Cfg, hs.AccessControl), hs.Index)
+	r.Get("/plugins/:id/", middleware.CanAdminPlugins(hs.Cfg, hs.AccessControl), hs.Index)
+	r.Get("/plugins/:id/edit", middleware.CanAdminPlugins(hs.Cfg, hs.AccessControl), hs.Index) // deprecated
+	r.Get("/plugins/:id/page/:page", middleware.CanAdminPlugins(hs.Cfg, hs.AccessControl), hs.Index)
 
 	r.Get("/connections/datasources", authorize(datasources.ConfigurationPageAccess), hs.Index)
 	r.Get("/connections/datasources/new", authorize(datasources.NewPageAccess), hs.Index)
 	r.Get("/connections/datasources/edit/*", authorize(datasources.EditPageAccess), hs.Index)
 	r.Get("/connections", authorize(datasources.ConfigurationPageAccess), hs.Index)
 	r.Get("/connections/add-new-connection", authorize(datasources.ConfigurationPageAccess), hs.Index)
-	r.Get("/connections/datasources/:id", middleware.CanAdminPlugins(hs.Cfg), hs.Index)
```

Comment:
Having only the plugin middleware here seems a bit suspicious to me :thinking:
 
Isn't it going to prevent users from configuring data sources that are not related to a plugin?
Isn't it also going to allow users that don't have `datasources.ConfigurationPageAccess` to access the page?

Reply:
In an ideal world we should not have to check plugin permissions when dealing with datasources.

Reply:
Issue that introduced this change: https://github.com/grafana/grafana/pull/61397

Reply:
So this is basically the same page as plugin catalog showing the details of a plugin so think it make sense to have it as is. See below

Connections/Add new connection: 
![image](https://github.com/grafana/grafana/assets/1668778/a2951a92-81d4-4646-ac50-202486ba7624)

Administration/Plugins:
![image](https://github.com/grafana/grafana/assets/1668778/fea996ff-17c3-4e32-a4b5-8142d3f9ef4f)

> Isn't it going to prevent users from configuring data sources that are not related to a plugin?

Plugins and datasources are different and thereby different permissions or am I misunderstanding something?

Reply:
@marefr is correct, this route is actually showing the "plugin details" page for a datasource plugin, and the settings of a datasource are available on this route:

```go
r.Get("/connections/datasources/edit/*", authorize(datasources.EditPageAccess), hs.Index)
```

I think it should be fine as it is, however I do agree that it's misleading - maybe we can add a comment above it?

Reply:
Ahhhh nice, thanks for the explanation! Really thought this was the edit page!
I'll add a comment :)